staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -355,7 +355,7 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 		switch {
 		case err != nil:
 			errorChan <- fmt.Errorf("error reading from error stream for port %d -> %d: %v", port.Local, port.Remote, err)
-		case len(message) > 0:
+		case len(message) > 0 && !strings.Contains(string(message), "connection reset by peer"):
```

Comment:
your current connection will be closed anyway by the operating system, you'll need to create  a new one

Reply:
does this not exit anyway?
I can see that close(errorChan) is always called

.. but `waitForConnection` keeps running :thinking:

Reply:
Yes the errChan is closed and handleConnection() exits, but since <-errorChan returns nil then it will not close streamConn (see line 405), so waitForConnection will be able to continue listening for and accepting new connections.